<a href="https://colab.research.google.com/github/scaballero1018/programacion1/blob/main/parcial2(1).ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Parcial 2
Sergio Andres Caballero Hernandez
2260671

instrucion1

In [114]:
import numpy as np
import pandas as pd
np.random.seed(0)
datos1 = np.random.randint(143, 197, 50)
datos=pd.DataFrame({
    "datos": datos1
})



In [115]:
datos['datos']

,datos
0,187
1,190
2,196
3,143
4,146
5,146
6,182
7,152
8,162
9,164


In [116]:
datos

,datos
0,187
1,190
2,196
3,143
4,146
5,146
6,182
7,152
8,162
9,164


In [38]:
datos.describe()

,datos
count,50.000000
mean,167.740000
std,16.315524
min,143.000000
25%,155.250000
50%,166.000000
75%,181.750000
max,196.000000


interpretacion
contiene 50 elementos\
con un promedio de 167\
con una desviacion estandar ded 16.315524\
el minimo valor es 143.000000\
el 25% valor es 155.250000  (el 25% mas grade po decirlo asi teniendo al 100% como el valor mas grande )\
valor medio 166.000000\
el 75% es 181.750000\
y el maxiomo valor o mas grade es 	196.000000

In [126]:
import pandas as pd
x = pd.DataFrame(datos)
normalizar = lambda x: (x - x.mean()) / x.std()
resultado = normalizar(x)
print(resultado)

       datos
0   1.180471
1   1.364345
2   1.732093
3  -1.516347
4  -1.332473
5  -1.332473
6   0.874014
7  -0.964725
8  -0.351812
9  -0.229230
10  1.548219
11  0.690140
12 -0.106647
13 -1.148599
14 -0.045356
15 -0.045356
16 -0.780851
17 -1.455056
18  0.812723
19  0.874014
20 -0.106647
21  1.303053
22 -0.045356
23 -0.474395
24  0.751432
25  0.015936
26 -0.719560
27 -1.026017
28 -0.964725
29 -0.290521
30  1.609510
31 -0.535686
32  1.609510
33 -1.209891
34 -0.596977
35  1.364345
36 -1.516347
37 -0.413103
38  0.628849
39 -0.045356
40  1.486927
41  1.609510
42  0.261101
43 -0.351812
44 -0.351812
45 -0.658269
46  0.874014
47  0.444975
48 -1.455056
49 -0.964725


instrucion 2

In [39]:
import pandas as pd
df= pd.DataFrame({
    "nombre":["ana","camilo","carlos","sergio","maria"],
    "edad": [23,25,27,17,20],
    "estatura":[1.70,1.80,1.85,1.70,1.77],
    "peso":[50,70,60,55,80]
})
df

,nombre,edad,estatura,peso
0,ana,23,1.70,50
1,camilo,25,1.80,70
2,carlos,27,1.85,60
3,sergio,17,1.70,55
4,maria,20,1.77,80


In [56]:
df["nombre"]

,nombre
0,ana
1,camilo
2,carlos
3,sergio
4,maria


In [58]:
df.loc[df['edad']>23]

,nombre,edad,estatura,peso
1,camilo,25,1.80,70
2,carlos,27,1.85,60


In [66]:
grupo = df.groupby("estatura")["peso"].mean()
grupo

,peso
estatura,
1.70,52.5
1.77,80.0
1.80,70.0
1.85,60.0


In [67]:
grupo1 = df.groupby("nombre")["edad"].mean()
grupo1

,edad
nombre,
ana,23.0
camilo,25.0
carlos,27.0
maria,20.0
sergio,17.0


In [112]:
from google.colab import files
import csv
datos = [
    ["Nombre", "Edad", "Estarura","Peso"],
    ["ana", 23, 1.70,50],
    ["camilo", 25, 1.80,70],
    ["carlos", 27, 1.85,60],
    ["sergio", 17, 1.70,55],
    ["maria", 20, 1.77,80]
]

with open("datos.csv", "w", newline="") as f:
    writer = csv.writer(f)
    writer.writerows(datos)
files.download("datos.csv")

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

parte 3

proyecto

In [70]:
import numpy as np
import pandas as pd

np.random.seed(123)

prueba_A = np.random.normal(loc=7.2, scale=0.4, size=50)
prueba_B = np.random.gamma(shape=306, scale=0.0229, size=50)

df = pd.DataFrame({
    "A": prueba_A,
    "B": prueba_B
})


In [71]:
df.describe()

,A,B
count,50.000000,50.000000
mean,7.205309,7.057722
std,0.480680,0.403559
min,6.080564,6.376485
25%,6.888402,6.740939
50%,7.153044,7.012551
75%,7.568949,7.329539
max,8.156946,8.092430


interpretacion
La prueba A contiene 50 elementos\
La prueba A tiene un promedio de 7.205309\
La prueba A tiene desviacion estandar de 16.315524\
La prueba A el minimo valor es 143.000000\
La prueba A el 25% valor es 155.250000  (el 25% mas grade po decirlo asi teniendo al 100% como el valor mas grande )\
La prueba A el valor medio 166.000000\
La prueba A el 75% es 181.750000\
En la prueba A el maxiomo valor o mas grade es 	196.000000

In [72]:
from scipy.stats import shapiro
shapiro(df["A"])

ShapiroResult(statistic=np.float64(0.98505924025743), pvalue=np.float64(0.774149900641557))

In [73]:
from scipy.stats import shapiro
shapiro(df["B"])

ShapiroResult(statistic=np.float64(0.9717389040641473), pvalue=np.float64(0.2720114624384205))

como el valor p es mayor que 0.05, no se rechaza la hipotesis de la normalidad con una confianza del 95%

In [93]:
from scipy import stats
stats.ttest_rel(df["A"], df["B"])


TtestResult(statistic=np.float64(1.729598919954534), pvalue=np.float64(0.08999751035350062), df=np.int64(49))

como el valor de p es mayor que 0,05 el puntaje medio de ambas pruebas es igual con una confianza del 95%


In [94]:
from scipy import stats
stats.kendalltau(df["A"], df["B"])

SignificanceResult(statistic=np.float64(0.04), pvalue=np.float64(0.6818954220320396))

como el valor de p es mayor que 0,05 el puntaje de las pruebas estan correlacionadas con una confianza del 95%

In [95]:
from scipy import stats
stats.spearmanr(df["A"], df["B"])

SignificanceResult(statistic=np.float64(0.07034813925570228), pvalue=np.float64(0.6273521677638413))

como el valor de p es mayor que 0,05 el puntaje de las pruebas estan correlacionadas con una confianza del 95%

In [136]:
import os
os.makedirs("comparacion", exist_ok=True)
%%writefile comparacion/pruebat.py:
import pandas as pd
import numpy as np
from scipy.stats import norm
from scipy import stats
prueba_A = np.random.normal(loc=7.2, scale=0.4, size=50)
prueba_B = np.random.gamma(shape=306, scale=0.0229, size=50)

np.random.seed(123)
prueba_A = np.random.normal(loc=7.2, scale=0.4, size=50)
prueba_B = np.random.gamma(shape=306, scale=0.0229, size=50)

df = pd.DataFrame({
    "A": prueba_A,
    "B": prueba_B
})
stats.ttest_rel(df["A"], df["B"])



UsageError: Line magic function `%%writefile` not found.


self en modulos, erorres, open ryas,

# CREAR UN CSV:
from google.colab import files
import csv
datos = [
    ["Nombre", "Edad", "Ciudad"],
    ["Ana", 25, "Bogotá"],
    ["Luis", 30, "Medellín"],
    ["Carlos", 28, "Cali"]
]

with open("datos.csv", "w", newline="") as f:
    writer = csv.writer(f)
    writer.writerows(datos)

files.download("datos.csv")

# CREAR MODULOS Y PAQUETES:

import os
os.makedirs("op", exist_ok=True)

%%writefile op/operaciones.py

# FUNCION LAMBDA
import pandas as pd
x = [1, 2, 3, 4]
x = pd.DataFrame(x)
normalizar = lambda x: (x - x.mean()) / x.std()
resultado = normalizar(x)
print(resultado)

